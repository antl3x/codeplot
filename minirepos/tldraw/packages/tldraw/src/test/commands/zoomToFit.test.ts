import { TestEditor, createDefaultShapes } from '../TestEditor'

let editor: TestEditor

beforeEach(() => {
	editor = new TestEditor()
	editor.createShapes(createDefaultShapes())
})

it('converts correctly', () => {
	editor.zoomToFit()
	expect({ ...editor.camera, id: 'static' }).toMatchSnapshot('Zoom to Fit Camera')
})

it('does not zoom to bounds when camera is frozen', () => {
	const cameraBefore = { ...editor.camera }
	editor.updateInstanceState({ canMoveCamera: false })
	editor.zoomToFit()
	expect(editor.camera).toMatchObject(cameraBefore)
})
