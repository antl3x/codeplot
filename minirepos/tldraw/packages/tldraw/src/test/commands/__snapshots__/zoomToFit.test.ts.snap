// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`converts correctly: Zoom to Fit Camera 1`] = `
Object {
  "id": "static",
  "meta": Object {},
  "typeName": "camera",
  "x": 330.435496777593,
  "y": 22.261531457640388,
  "z": 1.0469360106481882,
}
`;
